File: core/submission.go
package core

import (
	"github.com/louisevanderlith/husk/hsk"
	"github.com/louisevanderlith/husk/validation"
)

type Submission struct {
	Seller     Seller
	VehicleKey hsk.Key
	Price      float32
}

func (s Submission) Valid() error {
	return validation.Struct(s)
}

File: lib/bodies/submission.dart
import 'package:mango_ui/keys.dart';

import 'seller.dart';

class Submission {
  final Seller seller;
  final Key vehicleKey;

  Submission(this.seller, this.vehicleKey);

  Map<String, dynamic> toJson() {
    return {"Seller": seller, "VehicleKey": vehicleKey};
  }
}
